src/main/java/com/n1analytics/paillier/PaillierContext.java

```diff
@@ -362,22 +680,59 @@ public EncryptedNumber encrypt(EncodedNumber encoded) {
     return new EncryptedNumber(this, ciphertext, encoded.getExponent(), false);
   }
 
+  /**
+   * Encrypts a {@code Number}.
+   *
+   * @param value to be encrypted.
+   * @return the encryption result.
+   */
   public EncryptedNumber encrypt(Number value) {
     return encrypt(encode(value));
   }
 
+  /**
+   * Encrypts a {@code BigInteger}.
+   *
+   * @param value to be encrypted.
+   * @return the encryption result.
+   */
   public EncryptedNumber encrypt(BigInteger value) {
     return encrypt(encode(value));
   }
 
+  /**
+   * Encrypts a {@code double}.
+   *
+   * @param value to be encrypted.
+   * @return the encryption result.
+   */
   public EncryptedNumber encrypt(double value) {
     return encrypt(encode(value));
   }
 
+  /**
+   * Encrypts a {@code long}.
+   *
+   * @param value to be encrypted.
+   * @return the encryption result.
+   */
   public EncryptedNumber encrypt(long value) {
     return encrypt(encode(value));
   }
 
+  /**
+   * Adds two EncryptedNumbers. Checks whether the {@code PaillierContext} of {@code operand1}
+   * and {@code operand2} are the same as this {@code PaillierContext}. If the operands' exponents
+   * are not the same, reduce the higher exponent to match with the lower exponent. Throws
+   * PaillierContextMismatchException if the {@code PaillierContext} of either
```

Comment:
Don't need to repeat this statement in the summary. The `@throws` clause should say when the exception will be thrown.